modules/launcher/src/main/java/org/mule/module/launcher/AdaptedDeploymentListener.java

```diff
@@ -0,0 +1,104 @@
+ */
+package org.mule.module.launcher;
+
+
+import static java.lang.String.format;
+import static org.mule.config.bootstrap.ArtifactType.ALL;
+import static org.mule.config.bootstrap.ArtifactType.APP;
+import static org.mule.config.bootstrap.ArtifactType.DOMAIN;
+
+import org.mule.api.MuleContext;
+import org.mule.config.bootstrap.ArtifactType;
+
+import java.util.Arrays;
+
+
+/**
+ * Adapts an {@link ArtifactDeploymentListener} to work as a {@link DeploymentListener}.
+ */
+public class AdaptedDeploymentListener implements DeploymentListener
+{
+
+    private static final ArtifactType[] DEPLOYMENT_LISTENER_ARTIFACT_SUPPORTED_TYPES = {DOMAIN, APP};
+
+    public static final String UNSUPPORTED_ARTIFACT_TYPE_ERROR = format("DeploymentListener only supports %s artifact types.", Arrays.toString(DEPLOYMENT_LISTENER_ARTIFACT_SUPPORTED_TYPES));
+
+    private final ArtifactDeploymentListener artifactDeploymentListener;
+    private final ArtifactType artifactType;
+
+    public AdaptedDeploymentListener(ArtifactDeploymentListener artifactDeploymentListener, ArtifactType artifactType)
+    {
+        this.artifactType = validateArtifactType(artifactType);
+        this.artifactDeploymentListener = artifactDeploymentListener;
+    }
+
+    @Override
+    public void onDeploymentStart(String artifactName)
+    {
+        artifactDeploymentListener.onDeploymentStart(artifactType, artifactName);
+    }
+
+    @Override
+    public void onDeploymentSuccess(String artifactName)
+    {
+        artifactDeploymentListener.onDeploymentSuccess(artifactType, artifactName);
+    }
+
+    @Override
+    public void onDeploymentFailure(String artifactName, Throwable cause)
+    {
+        artifactDeploymentListener.onDeploymentFailure(artifactType, artifactName, cause);
+    }
+
+    @Override
+    public void onUndeploymentStart(String artifactName)
+    {
+        artifactDeploymentListener.onUndeploymentStart(artifactType, artifactName);
+    }
+
+    @Override
+    public void onUndeploymentSuccess(String artifactName)
+    {
+        artifactDeploymentListener.onUndeploymentSuccess(artifactType, artifactName);
+    }
+
+    @Override
+    public void onUndeploymentFailure(String artifactName, Throwable cause)
+    {
+        artifactDeploymentListener.onUndeploymentFailure(artifactType, artifactName, cause);
+    }
+
+    @Override
+    public void onMuleContextCreated(String artifactName, MuleContext context)
+    {
+        artifactDeploymentListener.onMuleContextCreated(artifactName, context);
+    }
+
+    @Override
+    public void onMuleContextInitialised(String artifactName, MuleContext context)
+    {
+        artifactDeploymentListener.onMuleContextInitialised(artifactName, context);
+    }
+
+    @Override
+    public void onMuleContextConfigured(String artifactName, MuleContext context)
+    {
+        artifactDeploymentListener.onMuleContextConfigured(artifactName, context);
+    }
+
+    private ArtifactType validateArtifactType(ArtifactType type)
```

Comment:
Remove this method. Use Preconditions#checkArgument. Validate that the type is not in the valid set of types